webui/react/src/services/decoder.ts

```diff
@@ -398,8 +395,11 @@ export const decodeTrialResponseToTrialDetails = (
     }));
   }
 
+  const EMPTY_STATES = new Set([ 'UNSPECIFIED', '', undefined ]);
+
   return {
     ...trialItem,
+    runnerState: EMPTY_STATES.has(data.trial.runnerState) ? undefined : data.trial.runnerState,
```

Comment:
**question:** not 100% sure but I think `data.trial.runnerState` should probably be `data.trial.state`?

Reply:
some extra typescript errors popped up after fixing this, but I ended up with something ugly like this lol:
```
  const EMPTY_STATES = new Set([ 'UNSPECIFIED', '', undefined ]);
  const trialState: string = data.trial.state as unknown as string;

  return {
    ...trialItem,
    runnerState: EMPTY_STATES.has(trialState) ? undefined : trialState,
    workloads: workloads || [],
  };
```

Reply:
**question:** can `data.trial.state` be something other than the enumerable values such as empty string or `undefined`? If not it can be some more simple:
```
  const isEmptyState = data.trial.state === Sdk.Determinedexperimentv1State.UNSPECIFIED;

  return {
    ...trialItem,
    runnerState: isEmptyState ? undefined : data.trial.state as unknown as string,
    workloads: workloads || [],
  };
```

Reply:
it can be an empty string for sure. but we shouldn't really get into undefined. we could at the decoder level make it undefined if it's in any of the empty states and that might make the usage easier.

Reply:
> question: not 100% sure but I think data.trial.runnerState should probably be data.trial.state?

wouldn't that get confusing with the existing trial state?

Reply:
What I meant was `data.trial` is of type `Trialv1Trial` which has `state` as a property but not `runnerState`:
![Screen Shot 2021-08-31 at 9 46 11 AM](https://user-images.githubusercontent.com/220971/131534376-2760b5ff-48b4-4660-bb17-4055cdff05d2.png)

Reply:
aah, this is expected this branch is missing the backend and api changes that are needed.

Reply:
could you check with the main push architecture branch see if that works fine for you? (the tests pass so at least should be fine)